Element84/filmdrop-aws-tf-modules
# Add cirrus lambdas

## Related issue(s)

- n/a

## Proposed Changes

1. Add lambdas to cirrus deployment
3. Remove unneeded cirrus IAM roles
4. Remove deprecated cirrus failed SNS topic
5. Rename stac-server `update-lambdas.sh` script to `update-stac-server-lambdas.bash`

Of the three lambdas with triggers (`api`, `process`, `update-state`), only the latter two have the triggers added here. The `api` is triggered off an API Gateway and that configuration has been omitted simply due to the level of effort required. That API Gateway configuration will need to be completed via future work.

## Testing

This change was validated by the following observations:

1. Deploy and check configurations of new lambdas
2. Try running test payloads into lambdas
3. Validate triggers have been created (where applicable)

Note that fully testing the lambdas is not currently possible due to the incomplete cirrus deployment. It might be worth having terraform optionally create S3 buckets and any other missing resources to allow the `flop` deployment to be testable.

## Checklist

- [x] I have deployed and validated this change
  - [x] I have added my changes to the changelog
  - [ ] No changelog entry is necessary
- [x] README migration
  - [ ] I have added any migration steps to the Readme
  - [x] No migration is necessary
